hudi-common/src/main/java/org/apache/hudi/common/fs/HoodieWrapperFileSystem.java

```diff
@@ -118,12 +156,31 @@ private static Registry getMetricRegistryForPath(Path p) {
     return executeFuncWithTimeMetrics(metricName, p, func);
   }
 
+  /**
+   * Executes the given function and updates the metric with time taken to execute the function and number of bytes.
+   *
+   * The number of bytes are returned from the execution of the provided function.
+   */
+  protected static int executeFuncWithTimeAndByteMetrics(String metricName, Path p, CheckedIntFunction func) throws IOException {
+    int ret = executeFuncWithTimeMetrics(metricName, p, func);
+    if (ret > 0) {
+      Registry registry = getMetricRegistryForPath(p);
+      if (registry != null) {
```

Comment:
The code invoke `getMetricRegistryForPath(p)` 2 times.

Reply:
Optimized.